src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java

```diff
@@ -185,23 +191,28 @@ private static void addAwsConfigGroup(final ConfigDef configDef) {
             ConfigDef.Width.NONE,
             AWS_S3_REGION_CONFIG
         );
+    }
+
+    private static void addFileConfigGroup(final ConfigDef configDef) {
+        int fileGroupCounter = 0;
 
         configDef.define(
-            AWS_S3_PREFIX_CONFIG,
-            Type.STRING,
+            FILE_NAME_TEMPLATE_CONFIG,
+            ConfigDef.Type.STRING,
             null,
-            new ConfigDef.NonEmptyString(),
-            Importance.MEDIUM,
-            "Prefix for stored objects, e.g. cluster-1/",
-            GROUP_AWS,
-            awsGroupCounter,
-            ConfigDef.Width.NONE,
-            AWS_S3_PREFIX_CONFIG
+            new FilenameTemplateValidator(FILE_NAME_TEMPLATE_CONFIG),
+            ConfigDef.Importance.MEDIUM,
+            "The template for file names on S3. "
+                + "Supports `{{ variable }}` placeholders for substituting variables. "
+                + "Currently supported variables are `topic`, `partition`, and `start_offset` "
+                + "(the offset of the first record in the file). "
+                + "Only some combinations of variables are valid, which currently are:\n"
+                + "- `topic`, `partition`, `start_offset`.",
```

Comment:
Here (and in the GCS connector) another possible combination (`key` only) is missing.